### PR TITLE
Pull card details from Zuora on the paid to paid upgrade page

### PR DESCRIPTION
**Pull request by tomverran**
So essentially to break our dependency on Salesforce for stripe info we need to:

- stop reading any card information from there
- Use a subscription rather than a contact when checking payment status

This PR handles the former, I am tackling the latter on https://github.com/guardian/membership-frontend/tree/subscription-action-refiner

### CODE REVIEW

File: frontend/app/controllers/TierController.scala [hunk 1]
```diff
@@ -87,7 +87,7 @@ trait UpgradeTier extends StripeServiceProvider with CatalogProvider {
       PageInfo(initialCheckoutForm = formI18n, stripePublicKey = stripeKey)
     }
 
-    def fromFree(subscription: FreeSubscription, contact: Contact[FreeTierMember, NoPayment]): Future[Result] =
+    def fromFree(subscription: FreeSubscription, contact: Contact[_, _]): Future[Result] =
```
Comment (tomverran) [inline frontend/app/controllers/TierController.scala]: I'm using underscores here to force compile errors if you try to do anything with the type parameters as I want to remove them soon

**Comment by ostapneko** [comment]
:+1:

**Comment by tomverran** [comment]
Yep I'm going to remove `PaymentStatus` from Contact in a branch on membership-common when I go through replacing `PaidMemberAction` with `PaidSubscriptionAction` or similar

**Comment by ostapneko** [comment]
:+1: Provided the Contact loses its parametrisation on PaymentStatus right after

**Pull request merged**